Additional updates to ui & adding ability to choose download folder

sdhutchins opened the pull request:
Will address #6 and #7 

### Additions:
- Updated `resultMessage`

sdhutchins committed:
Merge pull request #1 from djalmaaraujo/master

Merge with my master

sdhutchins committed:
Merge pull request #4 from djalmaaraujo/initial-ui

Initial ui

Commit:
Updated README with more relevant information.

Commit:
Updated dist script in package.json

Commit:
Updated error message.

Commit:
Edited size of bounce/loading circles.

Commit:
Edited message in initial-config

sdhutchins committed:
Merge pull request #5 from djalmaaraujo/master

Merge @linusg's commits into sdhutchins master

sdhutchins committed:
Merge pull request #6 from sdhutchins/ui

Initial changes

Commit:
Updates to README.md

sdhutchins committed:
Merge pull request #7 from djalmaaraujo/master

Update my ui branch with master

sdhutchins committed:
Merge pull request #8 from sdhutchins/ui

Update master to ui

Commit:
Updated package.json with author, description, and addition of electron-toolkit.

Commit:
Updated README

Commit:
Updated playlist-input component by adding better message as well as getting file to download.

Commit:
Removed unused variable.

sdhutchins commented:
@djalmaaraujo @linusg I'm just going to leave this open as I make changes. I hope to finish tomorrow/Sunday afternoon. Took me a few days to get the hang of electron, but now, I feel better about it all.

linusg commented on package.json:
```diff
@@ -1,33 +1,94 @@
+    "description": "A GUI for the spotify-downloader project.",
+    "build": {
+        "appId": "com.electron.spotifube",
+        "copyright": "Copyright © 2017 [object Object]",
```
lol what's that? Copyright which belongs to a JS object?

sdhutchins replied:
Again, electron-toolkit generated all of this not me. Not ready to be merged.

linusg commented on package.json:
```diff
@@ -1,33 +1,94 @@
+        "electronVersion": "1.7.8",
+        "asar": true,
+        "directories": {
+            "app": "C:\\Users\\Shaurita\\Documents\\GitRepo\\spotifube",
```
Please keep local dev paths out of any files commited.

sdhutchins replied:
It's not ready to be merged. I'm testing electron-toolkit for building the app and it created all of these paths.

linusg commented on package.json:
```diff
@@ -1,33 +1,94 @@
+    "name": "spotifube",
+    "productName": "spotifube",
+    "author": {
+        "name": "Djalma Araújo"
```
Probably this is picky but I would rather add all three of us to the author field, as @djalmaaraujo is the _owner_ of the repo and by far not it's own _author_ :thinking:

sdhutchins replied:
fwiw, i'd rather not be "named" in the project. Not really concerned with credit

djalmaaraujo replied:
@linusg I don't care about my name, if you want to move the repo to your username I can transfer and you put yourself as an author.

linusg replied:
@djalmaaraujo No, please don't get me wrong. I don't want to claim ownership either, but this whole thing is like a collaboration from the beginning. We all work and worked on this, so I would prefer to not have a single name here but either all of us or what about a generic 'The Spotifube developers'?

sdhutchins replied:
@djalmaaraujo @linusg I will just leave it blank when I make my PR.

linusg commented on package.json:
```diff
@@ -1,33 +1,94 @@
+    "productName": "spotifube",
+    "author": {
+        "name": "Djalma Araújo"
+    },
```
In general, what's the point of indenting this file with four spaces? The tool it was generated by (`yarn init`) chose two and I would say that's fine.

sdhutchins replied:
I didnt indent this, electron toolkit automatically did.

linusg commented on package.json:
```diff
@@ -1,33 +1,94 @@
+    },
+    "description": "A GUI for the spotify-downloader project.",
+    "build": {
+        "appId": "com.electron.spotifube",
```
Um, not sure if we would do it like this as `electron.com` does not belong to anyone of us... It may be even a non-existing domain, so reversed something like `com.spotifube.app`

sdhutchins replied:
Also, not cleaned up or ready to be merged. I didn't write this. Electron-toolkit auto generated it.

linusg commented on src/renderer/components/playlist-input/index.js:
```diff
@@ -32,14 +34,14 @@ export class PlaylistInput extends React.Component {
     window.spotifyDownloader.execute(
       data => {
-        this.setState({resultMessage: data});
+        this.setState({resultMessage: data,});
```
What's the point of the additional comma?

sdhutchins replied:
None. It's not cleaned up or ready to be merged.

linusg commented on src/renderer/components/playlist-input/index.js:
```diff
@@ -48,10 +50,17 @@ export class PlaylistInput extends React.Component {
+          this.setState({
+            playlistFetched: true, 
+            fetchError: false,
+            resultMessage: splitMessage[1] + " songs are available to download in this playlist.",
```
Better use formatting like

    const awesome = 'AWESOME';
    const foo = `This is ${awesome}!`;

sdhutchins replied:
Thanks for this!

linusg commented on src/renderer/components/playlist-input/index.js:
```diff
@@ -48,10 +50,17 @@ export class PlaylistInput extends React.Component {
           });
         } else {
-          this.setState({playlistFetched: true, fetchError: false});
+          let splitMessage = this.state.resultMessage.split(" ");
```
Should be `const` I think. I would suggest the constant use of single quotes in JS and double in HTML.

sdhutchins replied:
Just saw you had used this as well and was trying to keep with same style. I still have a lot to learn about js.

djalmaaraujo replied:
Should be const and we should always use single quotes.

linusg commented on src/renderer/scss/main.scss:
```diff
@@ -95,10 +95,10 @@ input[placeholder] { text-overflow: ellipsis; }
+    height: 25px;
     background-color: $color-green;
-    border-radius: 5px;
+    border-radius: 25px;
```
Radius is half the width/height, so `12.5px` which becomes `13px` after rounding up. Even though a bigger radius technically works it has no affect.

sdhutchins replied:
It's splitting hairs from me. It's just bigger and I like it bigger, but not a priority

linusg commented on src/renderer/scss/main.scss:
```diff
@@ -108,7 +108,7 @@ input[placeholder] { text-overflow: ellipsis; }
 
     &.bounce3 {
-      animation-delay: 0.4s;
+      animation-delay: 0.6s;
```
Can't tell if this look better or worse, but let's see!

linusg commented on src/renderer/scss/main.scss:
```diff
@@ -134,10 +134,10 @@ input[placeholder] { text-overflow: ellipsis; }
-   bottom: 40px;
-   left: 40px;
-   right: 40px;
+   top: 20px;
```
Same as above, it's hard to purely imagine an animation from code only.

linusg commented on README.md:
```diff
@@ -1,24 +1,25 @@
 # Spotifube
 
-> A `GUI` for the https://github.com/ritiek/spotify-downloader/ project.
+A `GUI` for [@ritiek](https://github.com/ritiek)'s [`spotify-downloader`](https://github.com/ritiek/spotify-downloader/) project.
```
@ritiek
Are you fine with this kind of attribution, or should your name be left off and just the project's name, or...? This whole project is building up on work in large parts done by you so we should make sure you're happy with it. :smile:

ritiek replied:
Thanks! The link to the repository alone is enough to make me happy. :smile:
Do whatever seems okay to you guys, I won't mind at all.

linusg commented on README.md:
```diff
@@ -1,24 +1,25 @@
-Simply clone down this reposity, install dependencies, and get started on your application.
+### What is spotify-downloader
+
+[`spotify-downloader`](https://github.com/ritiek/spotify-downloader/) is a command-line tool that matches Spotify tracks/playlists to youtube videos and downloads them as mp3/m4a, with id3, album art, etc.
```
Still picky, but if we write Spotify in a proper way (capital S) we have to write YouTube in a proper way too (capital Y and T).

sdhutchins replied:
Yeah, just made a few edits to the existing Readme. The sentence was already here, I just moved it.

djalmaaraujo commented:
I will review asap, please don't merge before. Need to take a deep look.

sdhutchins commented:
None of this is ready to merge or be reviewed. I opened the PR to let you guys know what I was working on so you guys could monitor it.

sdhutchins commented:
When I finish, I'll at you guys to review and suggest changes as I'm sure there will be many.

sdhutchins commented:
I have yet to commit what Im writing for 6 and 7 so reviewing this is premature which is why I let you guys know when I'd be done/ready for this to be reviewed.

sdhutchins commented:
> Anyone except me still working on anything? 🤔

The main reason I opened a PR was because this is covering more than 1 "issue", and I wanted to make sure that @linusg could track the actual progress of my commits because I understand that he wasn't sure if anyone was working on anything (and I have been for some days).

I've only added maybe 10% of what I need to add to address #6  and #7.  I've added very little code so at the moment - there is nothing relevant to review. I also have not cleaned any code.

Again, my goal is to have something by this afternoon (it's 3:33 am here), and when I have things ready (I'm very meticulous), I'll give you two a heads up.

I intend to follow along with @linusg's code style - I hate disjointed code bases so I'm not going to refactor his code at all.

In terms of the build stuff/package.json, none of that will be like what it is now - unless I get the build to work on my machine (then it'll look as it should and not be based on hardcoded paths) - I've just been trying out these node modules.

Additionally, I'll do a write up (as I do in other projects I belong to) in the main PR message to note what I changed/added/removed so it's easy to follow along.
